### PR TITLE
WebVR 1.1 spec compatibility

### PR DESCRIPTION
**Pull request by MortimerGoro**
Servo WebVR implementation started when WebVR spec 1.2 was about to be released. 1.2 API included some minor breaking changes from spec 1.1 in order to improve the support of the API in WebWorkers.

But eventually the WebVR  committee decided not to release 1.2 and make it a major version number with a lot more changes. WebVR API 2.0 is still under heavy churn.

This PR removes the WebVR changes that non-released 1.2 version introduced to support full WebVR 1.1 spec.

I pushed some GC fixes in a separate commit. See https://github.com/servo/servo/issues/17076 and https://github.com/servo/rust-mozjs/issues/351

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17608)

### CODE REVIEW

**Comment by KiChjang** [comment]
Is there a link to the specs you mentioned?

**Comment by MortimerGoro** [comment]
The link is not available anymore: https://github.com/w3c/webvr/pulls?q=is%3Aopen+is%3Apr+milestone%3A1.2

The changes were:
- Use navigator.vr.getDisplays instead of navigator.getVRDisplays()
- Use addEventListeners from VRDisplay objects instead of window.addEventListener

File: components/script/dom/vr.rs [hunk 1]
```diff
@@ -200,7 +200,7 @@ impl VR {
 
     fn notify_display_event(&self, display: &VRDisplay, event: &WebVRDisplayEvent) {
         let event = VRDisplayEvent::new_from_webvr(&self.global(), &display, &event);
-        event.upcast::<Event>().fire(self.upcast());
+        event.upcast::<Event>().fire(self.global().as_window().upcast::<EventTarget>());
```
Comment (jdm) [inline components/script/dom/vr.rs]: It should be possible to do `self.global().upcast::<EventTarget>()`.

File: components/script/dom/vrdisplay.rs [hunk 2]
```diff
@@ -467,7 +484,7 @@ impl VRDisplay {
     fn notify_event(&self, event: &WebVRDisplayEvent) {
         let root = Root::from_ref(&*self);
         let event = VRDisplayEvent::new_from_webvr(&self.global(), &root, &event);
-        event.upcast::<Event>().fire(self.upcast());
+        event.upcast::<Event>().fire(self.global().as_window().upcast::<EventTarget>());
```
Comment (jdm) [inline components/script/dom/vrdisplay.rs]: `self.global().upcast::`.

File: components/script/dom/webidls/VR.webidl [hunk 3]
```diff
@@ -3,7 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 // https://w3c.github.io/webvr/#interface-navigator
-[Pref="dom.webvr.enabled"]
+[NoInterfaceObject]
 interface VR: EventTarget {
```
Comment (jdm) [inline components/script/dom/webidls/VR.webidl]: Let's remove the parent interface, as well as from the source implementation.

**Review by jdm — CHANGES_REQUESTED** [review]

**Pull request merged**